src/Core/Tool/Authorizer/CountryCodeAuthorizer.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Ushahidi CountryCode Authorizer
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\Core\Tool\Authorizer;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\User;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Entity\Permission;
+use Ushahidi\Core\Tool\Authorizer;
+use Ushahidi\Core\Traits\AdminAccess;
+use Ushahidi\Core\Traits\UserContext;
+use Ushahidi\Core\Traits\PrivAccess;
+use Ushahidi\Core\Tool\Permissions\AclTrait;
+
+class CountryCodeAuthorizer implements Authorizer
+{
+	// The access checks are run under the context of a specific user
+	use UserContext;
+
+	// It uses `AdminAccess` to check if the user has admin access
+	use AdminAccess;
+
+	// It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+	use PrivAccess;
+
+	// Check that the user has the necessary permissions
+	// if roles are available for this deployment.
+	use AclTrait;
+
+	/* Authorizer */
+	public function isAllowed(Entity $entity, $privilege)
+	{
+		// These checks are run within the `User` context.
+		$user = $this->getUser();
+
+		// Allow role with the right permissions to do everything else
+		if ($this->acl->hasPermission($user, Permission::MANAGE_SETTINGS)) {
```

Comment:
I might be wrong, but shouldn't everyone be able to see a list of countries or a specific country?

Reply:
as in , I think isAllowed would probably always need to return true

Reply:
(for read/search , other methods should just be a 405)

Reply:
I don't know. It really doesn't matter if its available for everyone. I limited it since I wanted to keep it consistent with the permissions that the targeted surveys have.